client/dashboard/plugins/plugin/index.tsx

```diff
@@ -0,0 +1,263 @@
+import { TextBlur } from '../../components/text-blur';
+
+const decodeHtmlEntities = ( text: string ): string => {
+	const textarea = document.createElement( 'textarea' );
+	textarea.innerHTML = text;
+	return textarea.value;
+};
+
+const defaultSitesWithThisPluginView: View = {
+	type: 'table',
+	fields: [ 'activate', 'autoupdate', 'update' ],
+	sort: { field: 'name', direction: 'asc' },
+	titleField: 'domain',
+};
+
+const SitesWithThisPlugin = ( { sitesWithThisPlugin }: { sitesWithThisPlugin: Site[] } ) => {
+	const [ view, setView ] = useState< View >( defaultSitesWithThisPluginView );
+	const { pluginId } = pluginRoute.useParams();
+	const locale = useLocale();
+	const { data: sitesPlugins, isLoading: isLoadingSitesPlugins } = useQuery( pluginsQuery() );
+	const { isLoading: isLoadingSites } = useQuery( sitesQuery() );
+	const { isLoading: isLoadingWpOrgPlugin } = useQuery( wpOrgPluginQuery( pluginId, locale ) );
+
+	const fields = useMemo(
+		() => [
+			{
+				id: 'domain',
+				label: __( 'Site' ),
+				getValue: ( { item }: { item: Site } ) => item.URL,
+				render: ( { item }: { item: Site } ) => item.URL,
+				enableHiding: false,
+				enableSorting: true,
+				enableGlobalSearch: true,
+			},
+			{
+				id: 'activate',
+				label: __( 'Active' ),
+				getValue: ( { item }: { item: Site } ) =>
+					sitesPlugins?.sites[ item.ID ].find( ( p ) => p.slug === pluginId )?.active ?? false,
+				render: ( { item }: { item: Site } ) =>
+					sitesPlugins?.sites[ item.ID ].find( ( p ) => p.slug === pluginId )?.active ?? false,
+				enableHiding: false,
+				enableSorting: true,
+			},
+			{
+				id: 'autoupdate',
+				label: __( 'Autoupdate' ),
+				getValue: ( { item }: { item: Site } ) =>
+					sitesPlugins?.sites[ item.ID ].find( ( p ) => p.slug === pluginId )?.autoupdate ?? false,
+				render: ( { item }: { item: Site } ) =>
+					sitesPlugins?.sites[ item.ID ].find( ( p ) => p.slug === pluginId )?.autoupdate ?? false,
+				enableHiding: false,
+				enableSorting: true,
+			},
+			{
+				id: 'update',
+				label: __( 'Update' ),
+				render: () => 'Update',
+				enableHiding: false,
+				enableSorting: false,
+			},
+		],
+		[ pluginId, sitesPlugins ]
+	);
+
+	const { data, paginationInfo } = filterSortAndPaginate( sitesWithThisPlugin, view, fields );
+
+	return (
+		<DataViews
+			isLoading={ isLoadingSitesPlugins || isLoadingSites || isLoadingWpOrgPlugin }
+			data={ data }
+			fields={ fields }
+			view={ view }
+			onChangeView={ setView }
+			defaultLayouts={ { table: {} } }
+			actions={ [
+				{
+					id: 'delete',
+					label: __( 'Delete' ),
+					isPrimary: false,
+					callback: ( items ) => {
+						// Dummy delete action for now
+						// eslint-disable-next-line no-console
+						console.log( 'Delete clicked for plugin', items[ 0 ] );
+					},
+				},
+			] }
+			getItemId={ ( item ) => String( item.ID ) }
+			paginationInfo={ paginationInfo }
+		/>
+	);
+};
+
+const defaultSitesWithoutThisPluginView: View = {
```

Comment:
Could we move this so that it's closer to the other view definition?

Reply:
Extracted to its own component file instead.